idaes/models_extra/power_generation/unit_models/soc_submodels.py

```diff
@@ -0,0 +1,3997 @@
+                    "qflux_x0",
+                    idx_set=(tset,iznodes),
+                    units=pyo.units.W / pyo.units.m ** 2)
+    
+    _create_if_none(unit, 
+                    "qflux_x1",
+                    idx_set=(tset,iznodes),
+                    units=pyo.units.W / pyo.units.m ** 2)
+    
+def _material_boundary_conditions_config(CONFIG,thin):
+    if thin:
+        CONFIG.declare(
+            "xflux",
+            ConfigValue(default=None, description="Variable for material flux"),
+        )
+        CONFIG.declare(
+            "Dconc",
+            ConfigValue(default=None,
+                        description="Deviation of concentration from channel bulk "
+                        "concentration"),
+        )
+    else:
+        CONFIG.declare(
+            "Dconc_x0",
+            ConfigValue(default=None,
+                        description="Deviation of concentration at x= 0 from "
+                        "channel bulk concentration"),
+        )
+        CONFIG.declare(
+            "Dconc_x1",
+            ConfigValue(default=None,
+                        description="Deviation of concentration at x= 0 from "
+                        "channel bulk concentration"),
+        )
+        CONFIG.declare(
+            "xflux_x0",
+            ConfigValue(default=None, description="Variable for material flux at x=0"),
+        )
+        CONFIG.declare(
+            "xflux_x1",
+            ConfigValue(default=None, description="Variable for material flux at x=1"),
+        )
+        
+def _create_material_boundary_conditions_if_none(unit,thin):
+    tset = unit.flowsheet().config.time
+    iznodes = unit.iznodes
+    comps = unit.comps
+    if thin:
+        _create_if_none(unit, 
+                        "Dconc",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/pyo.units.m**3)
+        _create_if_none(unit, 
+                        "xflux",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/(pyo.units.s*pyo.units.m**2))
+    else:
+        _create_if_none(unit, 
+                        "Dconc_x0",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/pyo.units.m**3)
+        _create_if_none(unit, 
+                        "Dconc_x1",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/pyo.units.m**3)
+        _create_if_none(unit, 
+                        "xflux_x0",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/(pyo.units.s*pyo.units.m**2))
+        _create_if_none(unit, 
+                        "xflux_x1",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/(pyo.units.s*pyo.units.m**2))
+    
+@declare_process_block_class("SocChannel")
+class SocChannelData(UnitModelBlockData):
+    CONFIG = ConfigBlock()
```

Comment:
Is there a reason you are declaring a new `ConfigBlock` rather than inheriting the one from the base class? From what I can see, you are just replicating the things that are in the base class config anyway (`dynamic` and `has_holdup`).